doc/rm/c_cpp_coding_style.md

```diff
@@ -173,6 +186,58 @@ This may change if we find cases where this initialization improves readability.
 When initializing an array, initializers *may* be designated when that makes the array more readable (e.g., lookup tables that are mostly zeroed). Mixing designated and undesignated initializers, or using nested initializers, is still
 forbidden.
 
+### Nonstandard Attributes
+
+The following nonstandard attributes may be used:
+*   `section(<name>)` to put a definition into a given object section.
+*   `weak` to make a symbol definition have weak linkage.
+*   `naked` to ensure a function has no prolog or epilog.
+*   `interrupt` to ensure a function has the right prolog and epilog for interrupts (this involves saving and restoring more registers than normal).
+*   `packed` to ensure a struct contains no padding.
+
+It is recommended that other nonstandard attributes are not used, especially where C11 provides a standard means to accomplish the same thing.
+
+All nonstandard attributes must be supported by both GCC and Clang.
+
+Nonstandard attributes must be written before the declaration, like the following example, so they work with both declarations and definitions.
+
+```c
+__attribute__((section(".crt"))) void _crt(void);
```

Comment:
Can you please file a follow-up to get existing uses in the code base fixed to match this style?

Reply:
Yep, there'll be quite a significant PR which upgrades existing C code to C11. I'll start on that tomorrow.